### ISSUE TITLE
typescript parsing bug: 'export default abstract class C {}'

### ISSUE THREAD

**Issue by mjambon** [semgrep/semgrep#1995]
This requires fixing tree-sitter-typescript.

See https://github.com/tree-sitter/tree-sitter-typescript/issues/100 for details.

**Comment by baradm100** [comment]
Hi team!
I have this issue as well, any news?

## Logs

```
  --> /XXXXXXXXXXXXX/stateMachine.ts:3
3 | export default abstract class StateMachine {
  |                ^^^^^^^^
= help: If the code appears to be valid, this may be a semgrep bug.
Could not parse stateMachine.ts as ts
```

**Comment by mjambon** [comment]
We will be unblocked when this PR gets merged: https://github.com/tree-sitter/tree-sitter-javascript/pull/160

**Comment by chmccreery** [comment]
Following the link above, it looks like this is now unblocked - is that true @mjambon ? Are you up for assigning it to yourself or someone else on PA?

**Comment by mjambon** [comment]
@chmccreery correct and yes, all javascript and typescript parsing issues should be assigned to me.

**Comment by mjambon** [comment]
The fix will be in semgrep 0.56 due this week.
https://semgrep.dev/s/qrK7?version=develop now works
